[5.8] Corrects protocol when validating signature in UrlGenerator.php

agentmarine opened the pull request:
fixes #26834 where the URL being generated by $request->url() was not using the enforced url schema, swapped this to url()->current() which does use the enforced url schema.

this means that the signature generated would be valid with the one passed in.
EG: $request->url() generates http://www.example.com
       url()->current() generates https://www.example.com

agentmarine committed:
Update UrlGenerator.php

fixes #26834

GrahamCampbell commented on src/Illuminate/Routing/UrlGenerator.php:
```diff
@@ -355,7 +355,7 @@ public function temporarySignedRoute($name, $expiration, $parameters = [], $abso
     public function hasValidSignature(Request $request, $absolute = true)
     {
-        $url = $absolute ? $request->url() : '/'.$request->path();
+        $url = $absolute ? url()->current() : '/'.$request->path();
```
We can't use this helper because it's only available in the foundation package surerly?

agentmarine replied:
Works perfectly within my environment, generates the correct URL and respects protocols. Maybe this is working because of dependency injection from the underlying framework?

driesvints replied:
It works perfectly because you're loading in the entire framework. People using the package individual will experience problems. Since you're already in the url generator you can simply do `$this->current()` I believe.

agentmarine replied:
that works prefectly @driesvints I've committed it back.

driesvints committed:
Update src/Illuminate/Routing/UrlGenerator.php

Adjusted to use the URL already within the UrlGenerator rather than accessing Facade thanks to @driesvints

Co-Authored-By: agentmarine <dan@omegawebs.co.uk>

agentmarine commented:
trying to get the unit test to validate now.

taylorotwell commented:
I am closing this pull request because it lacks sufficient explanation, tests, or both. It is difficult for us to merge pull requests without these things because the change may introduce breaking changes to the framework.

Feel free to re-submit your change with a **thorough** explanation of the feature and tests - integration tests are preferred over unit tests. Please include it's benefit to end users; the reasons it does not break any existing features; how it makes building web applications easier, etc.

Thanks!
